tests/rust/wasm32-wasip3/src/bin/filesystem-mkdir-rmdir.rs

```diff
@@ -0,0 +1,103 @@
+use std::process;
+extern crate wit_bindgen;
+
+wit_bindgen::generate!({
+    inline: r"
+  package test:test;
+
+  world test {
+      include wasi:filesystem/imports@0.3.0-rc-2025-09-16;
+      include wasi:cli/command@0.3.0-rc-2025-09-16;
+  }
+",
+    additional_derives: [PartialEq, Eq, Hash, Clone],
+    // Work around https://github.com/bytecodealliance/wasm-tools/issues/2285.
+    features:["clocks-timezone"],
+    generate_all
+});
+
+use wasi::filesystem::types::Descriptor;
+use wasi::filesystem::types::ErrorCode;
+
+async fn test_mkdir_rmdir(dir: &Descriptor) {
+    let mkdir = |path: &str| dir.create_directory_at(path.to_string());
+    let rmdir = |path: &str| dir.remove_directory_at(path.to_string());
+
+    // create-directory-at: async func(path: string) -> result<_, error-code>;
+    assert_eq!(dir.create_directory_at("".to_string()).await,
+               Err(ErrorCode::NoEntry));
+    assert_eq!(mkdir(".").await, Err(ErrorCode::Exist));
+    assert_eq!(mkdir("..").await, Err(ErrorCode::NotPermitted));
+    assert_eq!(mkdir("parent/foo").await, Err(ErrorCode::NotPermitted));
+    assert_eq!(mkdir("/").await, Err(ErrorCode::NotPermitted));
+    assert_eq!(
+        mkdir("../fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(
+        mkdir("parent/fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(mkdir("a.txt").await, Err(ErrorCode::Exist));
+    mkdir("q.cleanup").await.unwrap();
+    assert_eq!(
+        rmdir("../fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(
+        rmdir("parent/fs-tests.dir/q.cleanup").await,
+        Err(ErrorCode::NotPermitted)
+    );
+    assert_eq!(rmdir("q.cleanup/").await, Err(ErrorCode::Invalid));
```

Comment:
Test filesystem-mkdir-rmdir failed
  [exit_code] 0 == 3
STDOUT:

STDERR:

thread '<unnamed>' (1) panicked at src/bin/filesystem-mkdir-rmdir.rs:51:5:
assertion `left == right` failed
  left: Err(ErrorCode { code: 0, name: "access", message: "Permission denied, similar to `EACCES` in POSIX." })
 right: Err(ErrorCode { code: 11, name: "invalid", message: "Invalid argument, similar to `EINVAL` in POSIX." })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: failed to run main module `tests\rust\testsuite\wasm32-wasip3\filesystem-mkdir-rmdir.wasm`

Reply:
Fixed this one by not testing rmdir with trailing slashes; that is https://github.com/bytecodealliance/wasmtime/issues/11524.